src/main/python/solr_integration_test.py

```diff
@@ -0,0 +1,210 @@
+        "topic_path":"src/main/resources/topics-and-qrels/topics.robust04.txt",
+        "qrel_path":"src/main/resources/topics-and-qrels/qrels.robust04.txt"
+    }
+}
+
+# keyword_search works with trec_eval() to extract actual MAP, P_30 result from "this_is_temp_file.txt" file
+def keyword_search(keywords):
+    if (os.path.exists("this_is_temp_file.txt") == True):
+        with open("this_is_temp_file.txt", "r") as f:
+            content = f.read().splitlines()   
+        res = []
+        for keyword in keywords:
+            for n in content:
+                if (keyword in n):res.append(n)
+    print(res) 
+    return res
+
+
+class Solr_command:
+    def __init__(self,name,input_path):
+        if (name == None): raise Exception("Need collection name, but receive None")
+        if (input_path == None): raise Exception("Need path of collection, but receive None")
+        self.name = name 
+        self.input_path = input_path
+        self.output_path = "run.solr.{}.bm25.topics.{}.301-450.601-700.txt".format(name,name)       
+
+    def command_execution(self, command_path, command):
+        print(command)
+        if (os.path.exists(command_path) == False):
+            sys.exit("default solr path: {} does not exist! Cannot execute the command. Please check:\n \
+            1.Solr is installed under correct path \
+            2.You run this program under correct path".format(command_path))    
+        output = subprocess.run(command, shell = True)
+
+    
+    def turn_on_server(self):    
+        print("we turn on solr server")
+        #### we check whether we have correct solr path first
+        solr_on = "solrini/bin/solr restart -c -m 8G"
+        solr_config = "pushd src/main/resources/solr && ./solr.sh ../../../../solrini localhost:9983 && popd"
+        self.command_execution("solrini/bin/solr", solr_on)
+        self.command_execution("src/main/resources/solr", solr_config)  
+        print("Solr server is on")
+
+
+    def turn_off_server(self):
+        subprocess.run("solrini/bin/solr stop -all",shell=True)        
+
+    # we consider 9983 as default local port number
+    # if this collection already exists on solr and has correct number of indexed document from previous test,we ask user 
+    # whether overwrite it, final MAP, P_30 result will be different for some reason. Thus we delete it and re-build it   
+    def indexing(self,name,input_path,thread_num):
+        thread_num = str(thread_num)
+        # if this collection already exists
+        collection_location = "solrini/server/solr/" + name + "_shard1_replica_n1"  
+        # if check_indexing return False, it mean previous indexed doc is incorrect,then we should overwrite this collection
+        # re-execute indexing, searching step.
+        if (os.path.exists(collection_location) and self.check_indexing() ):
+            response = input("collection already exists on Solr server and indexed doc number is same as expectancy\n \
```

Comment:
this script should not be interactive - this should be controlled via a command line option.